File: FFI/Controllers/retrieveMasterDefinitionListModel.cs
using System;
using System.Xml;
using System.Xml.Serialization;
using System.Xml.Schema;
using System.Collections.Generic;
using System.Web;

namespace MasterCodes.Models
{
    public partial class MasterCodesModels
    {
        #region retrieveMasterDefinitionList CLASSES 
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListRequest", Namespace="/retrieveMasterDefinitionList/Request")]
        public class retrieveMasterDefinitionListRequest
        {
            private retrieveMasterDefinitionListRequestcontext contextElement;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public retrieveMasterDefinitionListRequestcontext context
            {
                get
                {
                    return this.contextElement;
                }
                set
                {
                    this.contextElement = value;
                }
            }
            
        }
        
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListRequestcontext", Namespace="/retrieveMasterDefinitionList/Request")]
        public class retrieveMasterDefinitionListRequestcontext
        {
            private String orgnIdField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String orgnId
            {
                get
                {
                    return this.orgnIdField;
                }
                set
                {
                    this.orgnIdField = value;
                }
            }
            
            private String locnIdField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String locnId
            {
                get
                {
                    return this.locnIdField;
                }
                set
                {
                    this.locnIdField = value;
                }
            }
            
            private String userIdField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String userId
            {
                get
                {
                    return this.userIdField;
                }
                set
                {
                    this.userIdField = value;
                }
            }
            
            private String localeIdField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String localeId
            {
                get
                {
                    return this.localeIdField;
                }
                set
                {
                    this.localeIdField = value;
                }
            }
            
            public retrieveMasterDefinitionListRequestcontext() {}
            public retrieveMasterDefinitionListRequestcontext(String orgnId, String locnId, String userId, String localeId)
            {
                orgnIdField = orgnId;
                locnIdField = locnId;
                userIdField = userId;
                localeIdField = localeId;
            }
            private retrieveMasterDefinitionListRequestHeader HeaderElement;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public retrieveMasterDefinitionListRequestHeader Header
            {
                get
                {
                    return this.HeaderElement;
                }
                set
                {
                    this.HeaderElement = value;
                }
            }
            
        }
        
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListRequestHeader", Namespace="/retrieveMasterDefinitionList/Request")]
        public class retrieveMasterDefinitionListRequestHeader
        {
            private String parent_codeField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String parent_code
            {
                get
                {
                    return this.parent_codeField;
                }
                set
                {
                    this.parent_codeField = value;
                }
            }
            
            public retrieveMasterDefinitionListRequestHeader() {}
            public retrieveMasterDefinitionListRequestHeader(String master_code)
            {
                parent_codeField = master_code;
            }
        }
        
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListResponse", Namespace="/retrieveMasterDefinitionList/Response")]
        public class retrieveMasterDefinitionListResponse
        {
            private retrieveMasterDefinitionListResponsecontext contextElement;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public retrieveMasterDefinitionListResponsecontext context
            {
                get
                {
                    return this.contextElement;
                }
                set
                {
                    this.contextElement = value;
                }
            }
            
            private retrieveMasterDefinitionListResponseApplicationException ApplicationExceptionElement;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public retrieveMasterDefinitionListResponseApplicationException ApplicationException
            {
                get
                {
                    return this.ApplicationExceptionElement;
                }
                set
                {
                    this.ApplicationExceptionElement = value;
                }
            }
            
        }
        
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListResponseApplicationException", Namespace="/retrieveMasterDefinitionList/Response")]
        public class retrieveMasterDefinitionListResponseApplicationException
        {
            private String errorNumberField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String errorNumber
            {
                get
                {
                    return this.errorNumberField;
                }
                set
                {
                    this.errorNumberField = value;
                }
            }
            
            private String errorDescriptionField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String errorDescription
            {
                get
                {
                    return this.errorDescriptionField;
                }
                set
                {
                    this.errorDescriptionField = value;
                }
            }
            
        }
        
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListResponsecontext", Namespace="/retrieveMasterDefinitionList/Response")]
        public class retrieveMasterDefinitionListResponsecontext
        {
            private String orgnIdField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String orgnId
            {
                get
                {
                    return this.orgnIdField;
                }
                set
                {
                    this.orgnIdField = value;
                }
            }
            
            private String locnIdField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String locnId
            {
                get
                {
                    return this.locnIdField;
                }
                set
                {
                    this.locnIdField = value;
                }
            }
            
            private retrieveMasterDefinitionListResponseDetail[] DetailElement;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public retrieveMasterDefinitionListResponseDetail[] Detail
            {
                get
                {
                    return this.DetailElement;
                }
                set
                {
                    this.DetailElement = value;
                }
            }
            
        }
        
        [XmlTypeAttribute(TypeName="retrieveMasterDefinitionListResponseDetail", Namespace="/retrieveMasterDefinitionList/Response")]
        public class retrieveMasterDefinitionListResponseDetail
        {
            private int master_rowidField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public int master_rowid
            {
                get
                {
                    return this.master_rowidField;
                }
                set
                {
                    this.master_rowidField = value;
                }
            }
            

            private String parent_codeField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String parent_code
            {
                get
                {
                    return this.parent_codeField;
                }
                set
                {
                    this.parent_codeField = value;
                }
            }

         


            private String parent_descriptonField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String parent_descripton
            {
                get
                {
                    return this.parent_descriptonField;
                }
                set
                {
                    this.parent_descriptonField = value;
                }
            }

            private String master_codeField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String master_code
            {
                get
                {
                    return this.master_codeField;
                }
                set
                {
                    this.master_codeField = value;
                }
            }

            private String master_descriptionField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String master_description
            {
                get
                {
                    return this.master_descriptionField;
                }
                set
                {
                    this.master_descriptionField = value;
                }
            }
            
            

            private String master_ll_descriptionField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String master_ll_description
            {
                get
                {
                    return this.master_ll_descriptionField;
                }
                set
                {
                    this.master_ll_descriptionField = value;
                }
            }

            private String depend_descField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String depend_desc
            {
                get
                {
                    return this.depend_descField;
                }
                set
                {
                    this.depend_descField = value;
                }
            }
            
            private String status_descField;
            [XmlElementAttribute(Form = XmlSchemaForm.Unqualified)]
            public String status_desc
            {
                get
                {
                    return this.status_descField;
                }
                set
                {
                    this.status_descField = value;
                }
            }
            
        }
        
        #endregion
    }
}

File: FFI/wwwroot/API_Scripts/retrieveMasterDefinitionList_Model.js
var responseDataFormat = "JSON";

function retrieveMasterDefinitionListRequest(data)
{
    /**
     * @member {retrieveMasterDefinitionListRequestcontext} context
     */
    this['context'] = null;
    
    if(data)
    {
        if(data.hasOwnProperty('context') && data['context'] != null)
            this['context'] = new retrieveMasterDefinitionListRequestcontext(data['context']);
    }
}

function retrieveMasterDefinitionListRequestcontext(data)
{
    /**
     * @member {String} orgnId
     */
    this['orgnId'] = null;
    
    /**
     * @member {String} locnId
     */
    this['locnId'] = null;
    
    /**
     * @member {String} userId
     */
    this['userId'] = null;
    
    /**
     * @member {String} localeId
     */
    this['localeId'] = null;
    
    /**
     * @member {retrieveMasterDefinitionListRequestHeader} Header
     */
    this['Header'] = null;
    
    if(data)
    {
        if(data.hasOwnProperty('orgnId'))
            this['orgnId'] = data['orgnId'];
        
        if(data.hasOwnProperty('locnId'))
            this['locnId'] = data['locnId'];
        
        if(data.hasOwnProperty('userId'))
            this['userId'] = data['userId'];
        
        if(data.hasOwnProperty('localeId'))
            this['localeId'] = data['localeId'];
        
        if(data.hasOwnProperty('Header') && data['Header'] != null)
            this['Header'] = new retrieveMasterDefinitionListRequestHeader(data['Header']);
    }
}

function retrieveMasterDefinitionListRequestHeader(data)
{
    /**
     * @member {String} parent_code
     */
    this['parent_code'] = null;
    
    if(data)
    {
        if(data.hasOwnProperty('parent_code'))
            this['parent_code'] = data['parent_code'];
        
    }
}

function retrieveMasterDefinitionListResponse(data, format)
{
    /**
     * @member {retrieveMasterDefinitionListResponsecontext} context
     */
    this['context'] = null;
    
    /**
     * @member {retrieveMasterDefinitionListResponseApplicationException} ApplicationException
     */
    this['ApplicationException'] = null;
    
    if(format)
        responseDataFormat = format;
    
    if(data)
    {
        if (responseDataFormat == "XML") {
            if(data.getElementsByTagName("context").length > 0)
                this['context'] = new retrieveMasterDefinitionListResponsecontext(data.getElementsByTagName("context")[0]);
            
            if(data.getElementsByTagName("ApplicationException").length > 0)
                this['ApplicationException'] = new retrieveMasterDefinitionListResponseApplicationException(data.getElementsByTagName("ApplicationException")[0]);
        }
        else {
            if(data.hasOwnProperty('context') && data['context'] != null)
                this['context'] = new retrieveMasterDefinitionListResponsecontext(data['context']);
            
            if(data.hasOwnProperty('ApplicationException') && data['ApplicationException'] != null)
                this['ApplicationException'] = new retrieveMasterDefinitionListResponseApplicationException(data['ApplicationException']);
        }
    }
}

function retrieveMasterDefinitionListResponseApplicationException(data)
{
    /**
    * @member {String} sessionId
    */
    this['errorNumber'] = "";
    
    /**
    * @member {String} sessionId
    */
    this['errorDescription'] = "";
    
    if(data)
    {
        if (responseDataFormat == "XML") {
            if(data.getElementsByTagName("errorNumber")[0].lastChild != null)
                this['errorNumber'] = data.getElementsByTagName("errorNumber")[0].lastChild.nodeValue;
            
            if(data.getElementsByTagName("errorDescription")[0].lastChild != null)
                this['errorDescription'] = data.getElementsByTagName("errorDescription")[0].lastChild.nodeValue;
        }
        else {
            this['errorNumber'] = data['errorNumber'];
            this['errorDescription'] = data['errorDescription'];
        }
    }
}

function retrieveMasterDefinitionListResponsecontext(data)
{
    /**
     * @member {String} orgnId
     */
    this['orgnId'] = null;
    
    /**
     * @member {String} locnId
     */
    this['locnId'] = null;
    
    /**
     * @member {retrieveMasterDefinitionListResponseDetail} Detail
     */
    this['Detail'] = null;
    
    if(data)
    {
        if (responseDataFormat == "XML") {
            this['orgnId'] = (data.getElementsByTagName("orgnId")[0].lastChild != null) ? data.getElementsByTagName("orgnId")[0].lastChild.nodeValue : "";
            
            this['locnId'] = (data.getElementsByTagName("locnId")[0].lastChild != null) ? data.getElementsByTagName("locnId")[0].lastChild.nodeValue : "";
            
            Detail = [];
            for(Detailchildi = 0; Detailchildi < data.getElementsByTagName("Detail").length; Detailchildi++)
            {
                Detail.push(new retrieveMasterDefinitionListResponseDetail(data.getElementsByTagName("Detail")[Detailchildi]));
            }
            if(data.getElementsByTagName("Detail").length > 0)
                this['Detail'] = Detail;
        }
        else {
            this['orgnId'] = data['orgnId'];
            
            this['locnId'] = data['locnId'];
            
            if(data.hasOwnProperty('Detail') && data['Detail'] != null)
            {
                Detail = [];
                for(Detailindex = 0; Detailindex < data['Detail'].length; Detailindex++)
                {
                    Detail.push(new retrieveMasterDefinitionListResponseDetail(data['Detail'][Detailindex]));
                }
                this['Detail'] = Detail;
            }
        }
    }
}

function retrieveMasterDefinitionListResponseDetail(data)
{
    /**
     * @member {Int32} master_rowid
     */
    this['master_rowid'] = null;
    
    /**
     * @member {String} parent_code
     */
    this['parent_code'] = null;
    
    /**
     * @member {String} parent_descripton
     */
    this['parent_descripton'] = null;
    
    /**
     * @member {String} master_code
     */
    this['master_code'] = null;
    
    /**
     * @member {String} master_description
     */
    this['master_description'] = null;
    
    /**
     * @member {String} master_ll_description
     */
    this['master_ll_description'] = null;
    
    /**
     * @member {String} depend_desc
     */
    this['depend_desc'] = null;
    
    /**
     * @member {String} status_desc
     */
    this['status_desc'] = null;
    
    if(data)
    {
        if (responseDataFormat == "XML") {
            this['master_rowid'] = parseInt(data.getElementsByTagName("master_rowid")[0].lastChild.nodeValue);
            
            this['parent_code'] = (data.getElementsByTagName("parent_code")[0].lastChild != null) ? data.getElementsByTagName("parent_code")[0].lastChild.nodeValue : "";
            
            this['parent_descripton'] = (data.getElementsByTagName("parent_descripton")[0].lastChild != null) ? data.getElementsByTagName("parent_descripton")[0].lastChild.nodeValue : "";
            
            this['master_code'] = (data.getElementsByTagName("master_code")[0].lastChild != null) ? data.getElementsByTagName("master_code")[0].lastChild.nodeValue : "";
            
            this['master_description'] = (data.getElementsByTagName("master_description")[0].lastChild != null) ? data.getElementsByTagName("master_description")[0].lastChild.nodeValue : "";
            
            this['master_ll_description'] = (data.getElementsByTagName("master_ll_description")[0].lastChild != null) ? data.getElementsByTagName("master_ll_description")[0].lastChild.nodeValue : "";
            
            this['depend_desc'] = (data.getElementsByTagName("depend_desc")[0].lastChild != null) ? data.getElementsByTagName("depend_desc")[0].lastChild.nodeValue : "";
            
            this['status_desc'] = (data.getElementsByTagName("status_desc")[0].lastChild != null) ? data.getElementsByTagName("status_desc")[0].lastChild.nodeValue : "";
        }
        else {
            this['master_rowid'] = data['master_rowid'];
            
            this['parent_code'] = data['parent_code'];
            
            this['parent_descripton'] = data['parent_descripton'];
            
            this['master_code'] = data['master_code'];
            
            this['master_description'] = data['master_description'];
            
            this['master_ll_description'] = data['master_ll_description'];
            
            this['depend_desc'] = data['depend_desc'];
            
            this['status_desc'] = data['status_desc'];
        }
    }
}

